### PR TITLE
docs: organize local change testing info

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
- Organize and supplement the info over testing locally the changes performed. 
- Make a copy of the local docs to the docs/dir to keep it in both until we are able to remove them. 

NOTE: Detailed explanation in the comments. 

**Motivation for the change:**
Website docs.

### CODE REVIEW

File: doc/dev/developer_guide.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ $ make test
 ```
 
 For more information on running testing and correctly configuring your environment,
-refer to the [`Running the Tests Locally`][running-the-tests] document.
+refer to the [`Running the Tests Locally`][running-the-tests] document. Also, to check hot to test the changes performed locally see the [`Testing changes Locally`][testing-changes-locally] document.
```
Comment (camilamacedo86) [inline doc/dev/developer_guide.md:41]: To split the info and we have a doc describing how to test locally under the Testing category.

File: website/content/en/docs/contribution-guidelines/testing/local-changes.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+linkTitle: Local Changes
+weight: 40
+---
+
+## Testing Changes Locally
+
+If your changes are in the SDK commands then you just need to run `make install` to be able to use an SDK binary built from the source code and then test locally it. Also, see that you can run `operator-sdk version` to check what is the commit used to built it. 
+
+However, If the change performed is NOT in the scaffold files or sdk commands then, is required to build an new image with the changes done to test it locally in a POC operator project. In this way, by using this dev image in an operator project locally we will be able to check the changes made for the Ansible/Helm based-operator. And then, for the GO based-operators, will be required ensure that you are import your version of code implementation to be used in the POC operator project were it will be checked.  
+
```
Comment (camilamacedo86) [inline website/content/en/docs/contribution-guidelines/testing/local-changes.md:12]: Supplement the info by adding GO as well.

File: website/content/en/docs/contribution-guidelines/testing/local-docs.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ---
 title: Build and serve the docs locally
 linkTitle: Local Docs
+weight: 50
```
Comment (camilamacedo86) [inline website/content/en/docs/contribution-guidelines/testing/local-docs.md:4]: Move to the Testing Category

File: website/content/en/docs/contribution-guidelines/testing/local-changes.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	...
+	github.com/operator-framework/operator-sdk v0.0.0
+	...
+)
+
+// # Add a replace to the fork and branch with the changes
+replace github.com/operator-framework/operator-sdk => github.com/<fork>/operator-sdk <branch>
+```
+
+[makefile]: ../../../../../../Makefile
```
Comment (asmacdo) [inline website/content/en/docs/contribution-guidelines/testing/local-changes.md:91]: In hugo, relative links are within the built html, so it is not possible to use a relative link to a code file. Instead, use a full link to a github repo.

File: website/content/en/docs/contribution-guidelines/testing/local-docs.md (hunk 3 above)
Comment (asmacdo) [reply website/content/en/docs/contribution-guidelines/testing/local-docs.md:4]: Personally I would not think to look in the general "testing" area for how to build docs locally. IMO building the docs locally should stay top level in contribution guidelines.

File: doc/dev/developer_guide.md [hunk 5]
```diff
@@ -37,7 +37,7 @@ $ make test
...
```
Comment (asmacdo) [inline doc/dev/developer_guide.md:40]:
```suggestion
refer to the [`Running the Tests Locally`][running-the-tests] document. Also, to check how to test the changes performed locally see the [`Testing changes Locally`][testing-changes-locally] document.
```

File: website/content/en/docs/contribution-guidelines/testing/local-changes.md (hunk 2 above)
Comment (asmacdo) [reply website/content/en/docs/contribution-guidelines/testing/local-changes.md:12]: I don't think this guide belongs in "testing". I would assume that "testing" means running tests and would be surprised that I needed to read in this area just to get my local code changes into my environment.

Comment (camilamacedo86) [reply website/content/en/docs/contribution-guidelines/testing/local-changes.md]: Oks. moved to the main menu with local docs

File: website/content/en/docs/contribution-guidelines/testing/local-docs.md (hunk 3 above)
Comment (camilamacedo86) [reply website/content/en/docs/contribution-guidelines/testing/local-docs.md]: reverted

File: website/content/en/docs/contribution-guidelines/testing/local-changes.md (hunk 4 above)
Comment (camilamacedo86) [reply website/content/en/docs/contribution-guidelines/testing/local-changes.md]: done

File: doc/dev/developer_guide.md (hunk 5 above)
Comment (camilamacedo86) [reply doc/dev/developer_guide.md]: Done.

**Pull request merged**